### PR TITLE
After editing a file, redirect to the show page

### PR DESCRIPTION
**Pull request by tingtingjh**

### CODE REVIEW

File: spec/controllers/generic_files_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      GenericFile.create do |gf|
+        gf.apply_depositor_metadata(user)
+        gf.title = ['test title']
+      end
+    end
+
+    context "when updating metadata" do
+      let(:update_message) { double('content update message') }
+      before do
+        allow(ContentUpdateEventJob).to receive(:new).with(generic_file.id, 'test_user1@example.com').and_return(update_message)
```
Comment (rotated8) [inline spec/controllers/generic_files_controller_spec.rb]: Travis is 🔴 because this line hardcodes the email. Use `user.email` instead.

Comment (tingtingjh) [inline spec/controllers/generic_files_controller_spec.rb]: Got you. But as the example of "spaw a content update event job" has been tested by Sufia itself, I am going to omit it.

Comment (rotated8) [inline spec/controllers/generic_files_controller_spec.rb]: Ok. The test should make sure that when the edit page is posted, it redirects to the show page IF the metadata is valid, and back to the edit page (with a flash) if the metadata is not.

Comment (tingtingjh) [inline spec/controllers/generic_files_controller_spec.rb]: Updated as you suggested.

**Pull request merged**